base/compiler/abstractinterpretation.jl

```diff
@@ -1101,9 +1106,10 @@ function typeinf_local(frame::InferenceState)
                     end
                 elseif hd === :inbounds || hd === :meta || hd === :simdloop
                 else
-                    t = abstract_eval(stmt, changes, frame)
+                    used = !isempty(frame.ssavalue_uses[pc])
+                    t = abstract_eval(stmt, changes, frame, used)
```

Comment:
I'd like to change this code to preserve the computed edges list always (recomputing that is by-far the single slowest part of inference/optimize), rather than passing more arguments around. We don't actually compute the `backedges` list from these edges until much later (the `store_backedges` function) – it would be nice to try to keep all of the logic for that function in that one place there.

Reply:
Yes, that sounds like a good idea. I don't love the extra argument either. Fortunately there's no rush on this since the impact is not huge.